digdag-core/src/main/java/io/digdag/core/workflow/WorkflowCompiler.java

```diff
@@ -231,22 +232,25 @@ else if (config.has("_type") || config.getKeys().stream().anyMatch(key -> key.en
                     .map(pair -> collect(Optional.of(tb), fullName, pair.getKey(), pair.getValue(), validator))
                     .collect(Collectors.toList());
 
-                if (config.get("_parallel", boolean.class, false)) {
-                    // _after: is valid only when parallel: is true
-                    Map<String, TaskBuilder> names = new HashMap<>();
+                ParallelControl pc = new ParallelControl(config);
+                Map<String, TaskBuilder> names = new HashMap<>();
+                if (pc.isParallel() && pc.getParallelLimit() == 0) {
```

Comment:
Maybe the following is slightly better while it changes how to handle negative figure for `limit` ?
```
    if (pc.isParallel()) {
        if (pc.getParallelLimit() > 0) {
            :
        }
        else {
            :
        }
    }
    else {
        :
    }
```

Reply:
Good point, updated 3a11951